application/lib/templates.dart

```diff
@@ -29,291 +306,34 @@ class TemplateLibrary extends StatelessWidget {
-              },
-            ),
-            AtomDescription(
-              identifier: 'door',
-              className: 'TDoor',
-              properties: <String, PropertyValue>{
-                'name': StringPropertyValue('door'),
-                'pattern': StringPropertyValue('(open:1 closed:2)* door/doors'),
-                'frontSide': AtomPropertyValuePlaceholder('door_front'),
-                'backSide': AtomPropertyValuePlaceholder('door_back'),
-                'mass': LiteralPropertyValue('tmPonderous'),
-                'size': LiteralPropertyValue('tsMassive'),
-              },
-            ),
-            AtomDescription(
-              identifier: 'door_front',
-              className: 'TDoorSide',
-              properties: <String, PropertyValue>{
-                'name': StringPropertyValue('side'),
-                'pattern': StringPropertyValue('flat? (front:1)? side/sides'),
-                'description': StringPropertyValue('the door is flat.'),
-              },
-            ),
-            AtomDescription(
-              identifier: 'door_back',
-              className: 'TDoorSide',
-              properties: <String, PropertyValue>{
-                'name': StringPropertyValue('side'),
-                'pattern': StringPropertyValue('flat? (back:1)? side/sides'),
-                'description': StringPropertyValue('the door is flat.'),
-              },
-            ),
-            AtomDescription(
-              identifier: 'threshold_floor',
-              className: 'TThresholdSurface',
-              properties: <String, PropertyValue>{
-                'name': StringPropertyValue('floor'),
-                'pattern': StringPropertyValue(
-                    'flat? (ground/grounds surface/surfaces floor/floors)@'),
-                'description': StringPropertyValue('The floor is flat.'),
-                'mass': LiteralPropertyValue('tmLudicrous'),
-                'size': LiteralPropertyValue('tsLudicrous'),
-              },
-            ),
-          ],
-          icon: Icon(Icons.sensor_door),
-        ),
-      ],
+      children: templates,
     );
   }
 }
 
+Atom createFromTemplate(BuildContext context, List<AtomDescription> template) {
+  assert(template.isNotEmpty);
+  final List<Atom> atoms = template
+      .map<Atom>((AtomDescription description) =>
+          description.create(AtomsDisposition.of(context)))
+      .toList();
+  Atom _lookupAtom(String identifier, {Atom? ignore}) {
+    final List<Atom> matches = atoms
+        .where((Atom atom) =>
+            atom != ignore && atom.identifier!.matches(identifier))
+        .toList();
+    assert(matches.length == 1,
+        'could not find unique $identifier; found $matches in $atoms');
+    return matches.single;
+  }
+
+  for (final Atom atom in atoms) {
+    atom.resolveIdentifiers(_lookupAtom);
+  }
+  AtomsDisposition.of(context).addAll(atoms);
+  return atoms.first;
+}
+
 class Blueprint extends StatelessWidget {
```

Comment:
do we want to split the widget from the header/icon/atoms struct? it's a bit weird that we use a widget to carry this information around other places that don't use the widgettiness.